src/lib.rs

```diff
@@ -4,21 +4,80 @@
 
 #![no_std]
 
+use hubpack::SerializedSize;
+use serde::{Deserialize, Serialize};
 use zerocopy::{AsBytes, FromBytes};
 
 pub const DUMP_MAGIC: [u8; 4] = [0x1, 0xde, 0xde, 0xad];
 pub const DUMP_UNINITIALIZED: [u8; 4] = [0xba, 0xd, 0xca, 0xfe];
 pub const DUMP_SEGMENT_PAD: u8 = 0x55;
+
+const DUMP_SEGMENT_ALIGN: usize = 4;
+const DUMP_SEGMENT_MASK: usize = DUMP_SEGMENT_ALIGN - 1;
+
 pub const DUMP_REGISTER_MAGIC: [u8; 2] = [0xab, 0xba];
+pub const DUMP_TASK_MAGIC: [u8; 2] = [0xda, 0xda];
 
 pub const DUMPER_NONE: u8 = 0xff;
 pub const DUMPER_EMULATED: u8 = 0xfe;
+pub const DUMPER_EXTERNAL: u8 = 1;
+pub const DUMPER_JEFE: u8 = 2;
 
-const DUMP_SEGMENT_ALIGN: usize = 4;
-const DUMP_SEGMENT_MASK: usize = DUMP_SEGMENT_ALIGN - 1;
+pub const DUMP_AGENT_NONE: u8 = 0;
+pub const DUMP_AGENT_JEFE: u8 = 1;
+pub const DUMP_AGENT_TASK: u8 = 2;
+pub const DUMP_AGENT_INVALID: u8 = 0xff;
+
+#[derive(
+    Copy, Clone, Debug, SerializedSize, Serialize, Deserialize, PartialEq,
+)]
+pub enum DumpAgent {
+    /// No dump agent set
+    None,
+
+    /// Dump agent is Jefe, acting to dump a single task
+    Jefe,
+
+    /// Dump agent is a dedicated task, acting to dump entire system
```

Comment:
I don't love `Jefe` versus `Task`; maybe rename this to `DumpType` and specify `SingleTask` / `WholeImage`?

Reply:
Amusingly, this was one of the things that was bothering me too, and I had come to almost the same nomenclature. Pushed this change...